modules/promotion/src/PromotionUsageInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Drupal\commerce_promotion;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_promotion\Entity\CouponInterface;
+use Drupal\commerce_promotion\Entity\PromotionInterface;
+
+/**
+ * Provides a way to check the usage of a promotion or coupon.
+ */
+interface PromotionUsageInterface {
+
+  /**
+   * Gets the usage for a promotion.
+   *
+   * @param \Drupal\commerce_promotion\Entity\PromotionInterface $promotion
+   *   The promotion.
+   *
+   * @return int
+   *   The usage.
+   */
+  public function getPromotionUsage(PromotionInterface $promotion);
+
+  /**
+   * Add a promotion usage record.
+   *
+   * @param \Drupal\commerce_promotion\Entity\PromotionInterface $promotion
+   *   The promotion.
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   */
+  public function addPromotionUsage(PromotionInterface $promotion, OrderInterface $order);
```

Comment:
I guess. Just felt weird since we can get the promotion from the coupon, but a single method for adding works for me